### ISSUE TITLE
parsing ambiguities not well localized

### ISSUE THREAD

**Issue by grosu** [kframework/k-legacy#182]
We are often reported warnings like this:

```
[Warning] Parser: Parsing ambiguity. Arbitrarily choosing the first.
1: Case ::= Exp "->" Exp 
   k -> ((goto := k) ; 0)
2: Case ::= Exp "->" Exp 
   k -> (goto := (k ; 0))
```

The two productions reported in the ambiguity are identical.  The ambiguity is not there, is in the RHS of the arrow construct, and it is between '_:=_ and '_;_.

**Comment by grosu** [comment]
Also, the "Arbitrarily choosing the first." message should go after the ambiguity report.  I think this working would look best like this:

```
[Warning] Parser: Parsing ambiguity:
1: Exp ::= Exp ";" Exp 
   ((goto := k) ; 0)
2: Case ::= Exp ":=" Exp 
   (goto := (k ; 0))
Arbitrarily choosing the first.
```

**Comment by dwightguth** [comment]
Is this closeable?

**Comment by radumereuta** [comment]
If I remember correctly, this issue was reported because of a bug in SDF that I had to workaround several times until I managed to fix it.

Since we are trying to move to a completely new implementation of the parser, I think this can be closed for now. We will revisit the ambiguity reporting problem when we get the new parser. No reason to spend time patching the old one.

I will close it for now.

**Comment by grosu** [comment]
Don't rush.  We have no evidence yet that we will have a new parser any time soon.  And even if we do, it will probably take awhile until it beats SDF on programs.  Our new parser will probably be great for parsing the semantics, but for programs we will likely still want to be able to generate SDF parsers until we have a reasonably efficient one on our own.  Really, I'm quite concerned with our slow progress on our parser.  I almost feel like we should drop it.  At this moment it is borderline in terms of risk management.

Grigore

Sent from phone

-------- Original message --------
From: Radu Mereuta
Date:02/23/2014 18:49 (GMT-06:00)
To: kframework/k
Cc: "Rosu, Grigore"
Subject: Re: [k] parsing ambiguities not well localized (#182)

If I remember correctly, this issue was reported because of a bug in SDF that I had to workaround several times until I managed to fix it.

Since we are trying to move to a completely new implementation of the parser, I think this can be closed for now. We will revisit the ambiguity reporting problem when we get the new parser. No reason to spend time patching the old one.

I will close it for now.
